stan/math/prim/meta/require_generics.hpp

```diff
@@ -840,6 +840,11 @@ template <template <class...> class ContainerCheck,
 using require_container_st = require_t<
     container_scalar_type_check_base<ContainerCheck, TypeCheck, Check...>>;
 
+template <template <class...> class ContainerCheck,
+          template <class...> class TypeCheck, class... Check>
+using require_not_container_st = require_not_t<
+    container_scalar_type_check_base<ContainerCheck, TypeCheck, Check...>>;
+
```

Comment:
This should have a test or two.

Reply:
Also what this does is not consistent with other requires that just accept scalar check as a parameter. To be consistent with other requires I would suggest to change it to:
```
template <template <class...> class TypeCheck, class... Check>
using require_not_container_st = require_not_t<
    container_scalar_type_check_base<is_container, TypeCheck, Check...>>;
```

Reply:
It needs to be written this way to be consistent with ```require_container_st```, which allows for the specification of the container type (e.g., ```require_container_st<is_std_vector, is_integral, T>```)

Reply:
[optional] I see. I still think we should make a separate type trait for `require_container_st<is_container, ...>`. Though to be honest I can't think of a good name for it.